File: Day1/2sum.py
# This solution uses 1 pass hash approch
class Solution:
    
    def twoSum(self, nums, target):
    
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        table = {}
        i = 0

        while i < len(nums):
            comp = target - nums[i]

            if(comp) in table:
                return [table.get(comp), i]

            table[nums[i]] = i
            i += 1

File: Day1/2sum.java
// This also uses 1 pass hash approch
class Solution {

    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> table = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++){
            int comp = target - nums[i];
            
            if (table.containsKey(comp))
                return new int[] {table.get(comp), i};
            
            table.put(nums[i], i);
        }
        
        throw new IllegalArgumentException("No solution");
   
    }
}
